### ISSUE TITLE
[BRD] SimpleBard Song Option

### ISSUE THREAD

**Issue by hokeenthong** [Nik-Potokar/XIVSlothCombo#119]
Hi, 

I tried this and the first song it apply is Mage's Ballad and after it wears off, it will sometimes play both Wanderer's Minute or Army Peon together. (I am using Noclippy and could have double weaved). Sometimes, it will play two songs at start as well in my opener. There are times it would play the songs one by one as well. 

The most powerful song is Wanderer's Minute and as such, can you prioritize this song and put it as the 1st song it applies? Currently, the ideal situation is 43s of Wanderer's Minute, ~34s of Mage's Ballad, and ~43s of Army Peon. I am not sure if it is possible to do this and apply the song in this order and timing but putting Wanderer's minute first then the 2 others songs after would also be good as long it will re-apply Wanderer's minute when it is up again.  

Thanks.

**Comment by ele-starshade** [comment]
I believe it's currently coded to over-write Army's Paeon as soon as another song is off cooldown, no matter the time left on Army's Paeon.

I'd be happy to look into tweaking it into this "optimal" behaviour, @Nik-Potokar, if you like? :)

**Comment by hokeenthong** [comment]
Ah, if it is the case, then it explains why it will play 2 songs back to back. Thank you.

**Comment by Nik-Potokar** [comment]
@ele-starshade  Sure if that is the optimal one then please yes <3

**Comment by ele-starshade** [comment]
@hokeenthong - This has now been implemented under the "SimpleSong" option. Let us know if it works for you :)

Thanks!

**Comment by hokeenthong** [comment]
Thank you very much for helping. Really appreciate it. 

When I was trying it, it was the same as before. It will always play Mage's Ballad first and then sometimes, it will play Wanderer's Minute right away. In some cases, it will play Mage's Ballad only at start, and then it will play Wanderer's Minute after Mage's Ballad runs out but will play Army Peon right away after Wanderer's Minute. This behavior does not always happens but happens frequently. 

If it does not happen, it will play the sequence in this way. Mage's Ballad for full duration > Wanderer's Minute for full duration > Army Peon until Mage's Ballad is up and it will play Mage's Ballad when it is available and will not wait for the full duration of Army Peon.

**Comment by ele-starshade** [comment]
I think Nik just pushed it, so you might need to update your addon in-game first, I may have jumped the gun on mentioning you! (Didn't expect you to reply so fast! :D)

**Comment by Nik-Potokar** [comment]
Yeah just pushed it!

Thank you for all the work <3 @ele-starshade

**Comment by goofy-somebody** [comment]
I have been testing this and it does work as needed, although the `songTimerInSeconds` could probably be increased by 1.

Additionally there are instances where it may not work and I will continue to test those, as there may need to be additional conditional checks added.

So far, so good!  <3 @ele-starshade

**Comment by hokeenthong** [comment]
@ele-starshade oh lol I think I was too excited at your wonderful work. I will test i t later. Thanks again!

**Comment by hokeenthong** [comment]
Tested it and the songs order is correct. However, it is still playing 2 songs right after another. Thus, playing two songs together.

**Comment by ele-starshade** [comment]
Interesting, I didn't notice that myself. Can you share which toggles you have enabled and I will try to debug that tonight :)

**Comment by goofy-somebody** [comment]
@ele-starshade Condemned has a PR up that enhances on what you provided, which has some changes that might help, but as for the 2 songs at the same time, might indeed be a toggle conflict, so I'll leave that up to you and @hokeenthong to work out  :)

**Comment by ele-starshade** [comment]
Thank you @goofy-somebody!

**Comment by Codemned** [comment]
I'm unable to recreate hokeenthong's song issue.

**Comment by hokeenthong** [comment]
<img width="260" alt="Bard Toggles" src="https://user-images.githubusercontent.com/25314344/148961378-c6a1671d-b8d9-4a09-a3ca-d2d80a6dc109.PNG">

My toggles are in the image attached. 

I tried it many times and it is still the same. I suspect that it is because I was pressing the button too quickly (I am using turbo on the button). When I pressed it slower, it seems to be ok.

**Comment by goofy-somebody** [comment]
@hokeenthong 
Uncheck everything above 'SimpleBard' as all that is part of 'SimpleBard'
then see if you have the issue

**Comment by goofy-somebody** [comment]
@hokeenthong 
It should like the attached screenshot
![image](https://user-images.githubusercontent.com/41055964/148966229-b289b70a-bef5-4692-bfa0-7537c3f4c691.png)

**Comment by hokeenthong** [comment]
Did the exact toggle and still have the same issue.

**Comment by ele-starshade** [comment]
@hokeenthong - Codemned just pushed a hotfix and Aki built and pushed the new version, would you be able to test again please? I'm a bit concerned about this "Turbo" thing you spoke about as I don't know how we would solve that issue, I've not heard of the plugin that does that multi-weaving stuff you spoke about.

**Comment by Nik-Potokar** [comment]
@hokeenthong Could be noClippy if you are using that and your queue timer is too high (That was an issue before with another case)

**Comment by hokeenthong** [comment]
Tested it and same issue. I decided to test any conflict with other plugin. NoClippy disable and still have the same problem. I even tested the "turbo" thing and still have the same issue. In the end, it was ReAction (https://github.com/UnknownX7/ReAction) that was causing it. I had a stack where it covers all action. Removed it and it works. I will play around with it to see how I can manage this. 

Thanks for everyone who helped on this. After I removed ReAction, the songs are working as intended.
